File: 242_valid_anagram/242_valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if s == t:
            return True
        if len(s) != len(t):
            return False
        hashTable = {}

        for i in range(len(s)):
            if s[i] not in hashTable:
                hashTable[s[i]] = 0
            if t[i] not in hashTable:
                hashTable[t[i]] = 0
            hashTable[s[i]] += 1
            hashTable[t[i]] -= 1

        for i in hashTable.keys():
            if hashTable[i] != 0:
                return False

        return True

File: 242_valid_anagram/242_valid_anagram.go
package validanagram

func isAnagram(s string, t string) bool {
	if s == t {
		return true
	}
	if len(s) != len(t) {
		return false
	}

	hashTable := make(map[byte]int)
	for i := 0; i < len(s); i++ {
		hashTable[s[i]]++
		hashTable[t[i]]--
	}
	for _, v := range hashTable {
		if v != 0 {
			return false
		}
	}
	return true
}
